src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.il

```diff
@@ -458,6 +458,26 @@
         ret
   } // end of method Unsafe::IsAddressLessThan
 
+  .method public hidebysig static bool IsNullRef<T>(!!T& source) cil managed aggressiveinlining
+  {
+        .custom instance void System.Runtime.Versioning.NonVersionableAttribute::.ctor() = ( 01 00 00 00 )
+        .maxstack 2
+        ldarg.0
+        ldc.i4.0
+        conv.u
+        ceq
+        ret
+  } // end of method Unsafe::IsNullRef
+
+  .method public hidebysig static !!T& NullRef<T>() cil managed aggressiveinlining
+  {
+        .custom instance void System.Runtime.Versioning.NonVersionableAttribute::.ctor() = ( 01 00 00 00 )
+        .maxstack 1
+        ldc.i4.0
+        conv.u
+        ret
+  } // end of method Unsafe::NullRef
```

Comment:
Are these required after the implementation in IL: https://github.com/dotnet/runtime/blob/d8d56b4564848a9f4bb33fbc6eb74b8feefc77d7/src/libraries/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/Unsafe.cs#L376-L409

Reply:
Those should be the internal versions that use JIT intrinsics, while the APIs in this PR would be the public ones in pure IL that go into the [`System.Runtime.CompilerServices.Unsafe`](https://www.nuget.org/packages/System.Runtime.CompilerServices.Unsafe/) NuGet package.

We need both versions, see:

https://github.com/dotnet/runtime/blob/d8d56b4564848a9f4bb33fbc6eb74b8feefc77d7/src/libraries/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/Unsafe.cs#L19-L20